starter-code/server.js

```diff
@@ -6,37 +6,45 @@ const express = require('express');
 const bodyParser = require('body-parser');
 const PORT = process.env.PORT || 3000;
 const app = express();
-const conString = '';// TODO: Don't forget to set your own conString
+const conString = `postgres://postgres:${process.env.PG_PASSWORD}@localhost:5432/`; // DONE:Don't forget to set your own conString
 const client = new pg.Client(conString);
 client.connect();
 client.on('error', function(error) {
   console.error(error);
 });
 
 app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({extended: true}));
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
 app.use(express.static('./public'));
 
 app.get('/new', function(request, response) {
-  response.sendFile('new.html', {root: './public'});
+  response.sendFile('new.html', {
+    root: './public'
+  });
 });
 
 app.get('/articles', function(request, response) {
   // REVIEW: This query will join the data together from our tables and send it back to the client.
-  // TODO: Write a SQL query which joins all data from articles and authors tables on the author_id value of each
-  client.query(``)
-  .then(function(result) {
-    response.send(result.rows);
-  })
-  .catch(function(err) {
-    console.error(err)
-  });
+  // DONE: Write a SQL query which joins all data from articles and authors tables on the author_id value of each
+  client.query(`SELECT * FROM articles INNER JOIN authors ON articles.author_id = author.author_id`)
+    .then(function(result) {
+      response.send(result.rows);
+    })
+    .catch(function(err) {
+      console.error(err)
+    });
 });
 
-app.post('/articles', function(request, response) {
+app.post('/articles', function(req, res) {
   client.query(
-    '', // TODO: Write a SQL query to insert a new author, ON CONFLICT DO NOTHING
-    [], // TODO: Add the author and "authorUrl" as data for the SQL query
+    'INSERT INTO authors (author_id, author, "authorUrl") VALUES ($1, $2, $3) ON CONFLICT DO NOTHING;', // DONE: Write a SQL query to insert a new author, ON CONFLICT DO NOTHING
```

Comment:
giving you the points for this...you didn't need author_id, though, since if it's new it'll be generated by sql